### PR TITLE
Add office table

### LINKED ISSUE #4: [4 pts]  Add Offices table
Build the related rooms table components including:
- Migration file
- DB validators
- Model
- Controller with the related end points:
- Index (Home page)
- Show (Details page)
- New (Add from
- Delete

### PR DESCRIPTION
**Pull request by Fourteen98**
### In This Pull Request: 
- Created Office migration file based on ERD. 
- Generated Schema
- Created office  model with validations
- Created office controllers with index, show, create and destroy actions

### CODE REVIEW

File: app/controllers/offices_controller.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class OfficesController < ApplicationController
+  def index
+    @offices = Office.all
+    render json: @offices
+  end
+
+  def show
+    @office = Office.find(params[:id])
+    render json: @Office
```
Comment (jaferIdrees) [inline app/controllers/offices_controller.rb]: You render an empty variable here, 'o is small-letter in line 3 but capital in line 4'

File: app/models/office.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Office < ApplicationRecord
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :area, presence: true
+  validates :occupancy, presence: true, numericality: { only_integer: true }
+  validates :images, presence: true
+  validates :basic_price, presence: true, numericality: true
+  validates :address, presence: true
+end
```
Comment (jaferIdrees) [inline app/models/office.rb:9]: [Optional] It's a good idea make all fields as required but sometimes it makes negative UX, i.e., area my considered as an alternative for the occupancy.

**Review by jaferIdrees — CHANGES_REQUESTED** [review]
Hi @Fourteen98 

Many thanks for your great efforts.
You satisfied the card's requirements in a professional way; you have a minor typo-mistake you need to correct it.
On the other hand, below I have one optional suggestion for you.

Keep hacking.

File: app/models/office.rb (hunk 2 above)
Comment (Fourteen98) [reply app/models/office.rb:9]: @jaferIdrees , can you elaborate on this well.

Comment (Fourteen98) [reply app/models/office.rb:9]: too much validations you mean right and that gives bad user experience?

**Review by jaferIdrees — APPROVED** [review]
Hi @Fourteen98 
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

Highlights
👍🏽 Implemented all requirements ✔️
👍🏽 PR looks professional ✔️
👍🏽 Linters installed and errors ✔️
👍🏽 Commits are professional ✔️

Cheers and Happy coding!👏👏👏

**Pull request merged**